app/controllers/admin/schools/induction_coordinators_controller.rb

```diff
@@ -16,13 +16,72 @@ def create
       @nominate_induction_tutor_form = NominateInductionTutorForm.new(
         form_params.merge(school_id: params[:school_id]),
       )
-      render :new and return unless @nominate_induction_tutor_form.valid?
 
-      @nominate_induction_tutor_form.save!
-      set_success_message(content: "New induction tutor added. They will get an email with next steps.", title: "Success")
-      redirect_to admin_school_path(@school)
-    rescue UserExistsError
-      redirect_to email_used_admin_school_induction_coordinators_path
+      if @nominate_induction_tutor_form.valid?
+        if email_already_used?(@nominate_induction_tutor_form.email)
+          @email_address = @nominate_induction_tutor_form.email
+          @another_school = school_using_this_email(@email_address)
+          render "email_used"
+        else
+          create_school_induction_tutor!(school: @school,
+                                         email: @nominate_induction_tutor_form.email,
+                                         full_name: @nominate_induction_tutor_form.full_name)
+          set_success_message(content: "New induction tutor added. They will get an email with next steps.", title: "Success")
+          redirect_to admin_school_path(@school)
+        end
+      else
+        render :new
+      end
+    end
+
+    def create_school_induction_tutor!(school:, email:, full_name:)
```

Comment:
Let's make as much stuff as possible `private`. I like the public methods in the controller to be just the actions - cause I don't think anything else needs to be.

I also don't like to have too much logic in the controller, and we happen to have a form object wandering around - maybe using that for some checks (like if the email is being used) would reduce the cognitive load on the controller?

Reply:
Another good catch, agree about the `private` stuff, I had meant to refactor that before submitting the PR.  Kind of agree about the controllers being lean but I wasn't too keen on the form object handling the save and raising an exception, I prefer to treat them like a simple view model. Additionally we need to pass additional info concerning where the email address is being used in the "error" page so needed to restructure it.

Reply:
The approach we've been taking here is to lift the majority of the logic in to the `form`s, which makes things easier to test. @BroiSatse is likely to have view on this. 

I personally _much_ prefer the logic in the form to in the controller